tests/security/grub_auth/grub_authorization.pm

```diff
@@ -39,26 +39,45 @@ my $maint_passwd = 'pw_maintainer';
 my $test_user    = 'tester';
 my $test_passwd  = 'pw_test';
 
+sub switch_boot_menu {
+    if (get_var('UEFI')) {
+        assert_screen("grub_uefi_firmware_menu_entry", timeout => 90);
+        send_key_until_needlematch("grub_auth_boot_menu_entry", "down", 5, 2);
+    }
+    else {
+        assert_screen("grub_auth_boot_menu_entry", timeout => 90);
+    }
+}
+
+sub check_boot_menu {
+    if (get_var('UEFI')) {
+        assert_screen("grub_uefi_firmware_menu_entry", timeout => 90);
+    }
+    else {
+        assert_screen("grub_auth_boot_menu_entry", timeout => 90);
+    }
+}
+
```

Comment:
There is one difference between `switch_boot_menu` and `check_boot_menu`, you could pass a parameter to control if do `send_key_until_needlematch("grub_auth_boot_menu_entry", "down", 5, 2);`, then you only need one sub function.

Reply:
Can not agree with you any more, let me fix it